docs2/site/docs/migrations/migration8.md

```diff
@@ -246,6 +246,80 @@ At this time GraphQL.NET does not directly support the `MaxLength` and similar a
+`IValidationRule` and `IVariableVisitorProvider`. The `IVariableVisitorProvider` interface has a
+new method `ValidateArgumentsAsync` that is called during validation. This method is called after
+all field arguments and directive arguments have been parsed, allowing a validation rule to validate
+parsed argument values. It will run even if there are no arguments to validate, so you should check
+for this case and return early if there are no arguments to validate. However it will not run if
+errors occur prior to parsing the arguments, such as if the field or directive is not found, or
+if an argument fails coercion or validation.
+
+Sample code for a validation rule that ensures that connections do not request over 1000 rows:
+
+```csharp
+public class NoConnectionOver1000ValidationRule : IValidationRule, IVariableVisitorProvider, INodeVisitor
+{
+    // do not run any visitors on the initial validation
+    public ValueTask<INodeVisitor?> ValidateAsync(ValidationContext context) => default;
+
+    // after argument parsing, run this rule only when there are argument values specified
+    ValueTask<INodeVisitor?> IVariableVisitorProvider.ValidateArgumentsAsync(ValidationContext context)
+        => context.ArgumentValues != null ? new(this) : default;
+
+    // IVariableVisitor is not used in this sample
+    IVariableVisitor? IVariableVisitorProvider.GetVisitor(ValidationContext context) => null;
+
+    // look for connection arguments and validate the value
+    ValueTask INodeVisitor.EnterAsync(ASTNode node, ValidationContext context)
+    {
+        // look for field nodes and find the field definition
+        if (node is not GraphQLField fieldNode)
+            return default;
+        var fieldDef = context.TypeInfo.GetFieldDef();
+        if (fieldDef == null)
+            return default;
+        // look for connection types
+        if (fieldDef.ResolvedType?.GetNamedType() is not IObjectGraphType connectionType || !connectionType.Name.EndsWith("Connection"))
+            return default;
+        // retrieve the arguments
+        if (!(context.ArgumentValues?.TryGetValue(fieldNode, out var args) ?? false))
+            return default;
+        // look for first and last arguments
+        ArgumentValue lastArg = default;
+        if (!args.TryGetValue("first", out var firstArg) && !args.TryGetValue("last", out lastArg))
+            return default;
+        var rows = (int?)firstArg.Value ?? (int?)lastArg.Value ?? 0;
+        if (rows > 1000)
+            context.ReportError(new ValidationError("Cannot return more than 1000 rows"));
+        return default;
+    }
+
+    ValueTask INodeVisitor.LeaveAsync(ASTNode node, ValidationContext context) => default;
+}
+```
+
+The above is sample functionality, although in this situation it would be more efficient to
+have a schema visitor apply this validation rule to the appropriate fields via the `Validate`
+method during schema initialization.
```

Comment:
Can you add an example here to clarify the difference between the two approaches?

Reply:
Done